src/wovnio/wovnphp/core/url_handlers/WovnPathURLHandler.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+
+namespace Wovnio\Wovnphp\Core\UrlHandler;
+
+use Wovnio\Wovnphp\Core\WovnLangException;
+use Wovnio\Wovnphp\Core\WovnOption;
+
+class WovnPathURLHandler extends WovnURLHandler
+{
+    public function __construct($original, $langDirectory, $options)
+    {
+        parent::__construct($original, $langDirectory, $options);
+    }
+
+    protected function detectLang()
+    {
+        $path = strval($this->components['path']);
+        $prefix = strval($this->options->get(WovnOption::OPT_SITE_PREFIX_PATH)) . '/';
+        $match = strpos($path, $prefix);
+        if ($match !== false) {
+            $path = substr($path, $match + strlen($prefix));
+        }
+        $exploded  = explode('/', $path);
+        if (count($exploded) === 1) {
+            return $this->langDirectory->defaultLang();
+        }
+        $langCandidate = $exploded[0];
+        try {
+            $lang = $this->langDirectory->getLang($langCandidate);
+            if (!$lang->isValidLang()) {
+                $lang = $this->langDirectory->defaultLang();
+            }
+        } catch (WovnLangException $e) {
+            $lang = $this->langDirectory->defaultLang();
+        }
+        return $lang;
+    }
+
+    protected function removeLang()
```

Comment:
This is OK to leave as stubs for now, but note that the `addLang/removeLang` approach is kind of  out of date after you take in account newer features like default language aliases, custom domain. 
Sometimes you need to "add" a code to the URL to arrive at the default language. 

We would do well to move towards `convertToDefaultLang` and `convertToTargetLang` which are the more higher level operations we actually mean.